src/Theme/utils.ts

```diff
@@ -0,0 +1,55 @@
+import { DefaultTheme } from 'styled-components';
+
+/**
+ * Helper function that returns a function that extracts values from the theme.
+ * Allows passing multiple strings as props to dig into nested values.
+ * This is intended for use in styled-comopnents's interpolation functions as
+ * an easier way to get properly typed values. Examples:
+ *
+ * instead of:
+ *
+ * styled.div`
+ *   margin-top: ${({theme}: {theme: DefaultTheme}):string => theme.ws.xsmall}
+ * `
+ *
+ * use instead:
+ *
+ * styled.div`
+ *   margin-top: ${fromTheme('ws', 'xsmall')}
+ */
+
+export const fromTheme = (...levels: string[]) => (
+  ({ theme }: { theme: DefaultTheme }): string => {
+    const propList = [...levels];
+    let currentLevel = theme;
+    while (propList.length > 0) {
+      const nextLevel = propList.shift();
```

Comment:
The airbnb guide actually [specifically prohibits iterators like `for ... of`](https://github.com/airbnb/javascript#iterators--nope).

Reply:
I thought about this (and the [other note below](https://github.com/seas-computing/mark-one/pull/69/files/7fd4fe55ddee4572ec66d945f4a5d72702a11428#diff-b685306b07e26c2e8be5b82704f151e7) some more and rewrote `fromTheme` in d8e03ccd. There's some TS trickery going on there to at least check that the first argument passed to the function is one of the top-level keys in Theme, but I think it would take a bit more work to get it to dynamically check types on the subsequent values.

Reply:
The way that I read the airbnb docs is that you should avoid iterators in general if you can use a higher-order function, so I guess the question is whether a `while` loop is better than a `for ... of` loop. (My understanding was they listed those two just as an example of iterators?)

I dug around the airbnb guide and found this in the issues: https://github.com/airbnb/javascript/issues/1439
One of the collaborators said that all loops are banned: 

![image](https://user-images.githubusercontent.com/29589689/88224109-8206b780-cc36-11ea-852e-e48800c4670a.png)

Reply:
Do you think we should leave the checking of the subsequent values for a later ticket?

Reply:
Re: the comments on loops, I think I see their point. Ideally, I would have used a `reduce()` function in place of a loop, but I was running into Typescript issues that I sort of sidestepped (or dove deeper into) by using a recursive function instead, and limited the depth to only 4 levels. I think the ultimate result in 6af3f48 is still pretty hacky, but it does offer better type safety.